CMakeLists.txt

```diff
@@ -16,7 +16,40 @@ catkin_package(
   )
 
 add_library(${PROJECT_NAME} src/realtime_clock.cpp)
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${Boost_LIBRARIES})
+target_link_libraries(${PROJECT_NAME} PUBLIC ${catkin_LIBRARIES} ${Boost_LIBRARIES})
+
+# Unit Tests
+if (CATKIN_ENABLE_TESTING)
+  catkin_add_gtest(realtime_box_tests test/realtime_box_tests.cpp)
+  if (TARGET realtime_box_tests)
```

Comment:
Please forgive my ignorance on this one but I forgot the why of this hacky IF setup. Can we avoid using it?

Reply:
I was copying [an example from the ros wiki](http://wiki.ros.org/gtest#Update_CMakeLists.txt). It looks like `catkin_add_gtest()` [always calls `add_executable()`](https://github.com/ros/catkin/blob/94c7d2e25ee2aa6b1f19a41f1deda731c7f8af63/cmake/test/gtest.cmake#L171) so this check shouldn't be necessary.

Reply:
It looks like rostest [only creates a target if gtest/gmock is found](https://github.com/ros/ros_comm/blob/29053c4832229efa7160fb944c05e3bc82e11540/tools/rostest/cmake/rostest-extras.cmake.em#L78).

Reply:
Removed all `if(TARGET ...)` checks in be439e97931e55d59d8500f310046f41572a1a54. If `gtest/gmock` isn't found then this code will complain at configure time.

Reply:
Thanks. I actually had an unsubmitted comment on this:

"Yes. And in the case they aren't, I think the compilation should stop, error out and potentially set the computer on fire. If there are tests and we set the flag to compile tests, they should be compiled."

Reply:
Thank you